src/bindings/python/src/pyopenvino/core/tensor.cpp

```diff
@@ -16,6 +16,42 @@ void regclass_Tensor(py::module m) {
     py::class_<ov::Tensor, std::shared_ptr<ov::Tensor>> cls(m, "Tensor");
     cls.doc() = "openvino.runtime.Tensor holding either copy of memory or shared host memory.";
 
+    // dtype for py::dtype and py::object
+    cls.def(py::init([](py::array& array, ov::element::Type& dtype, bool share) {
+                if (dtype != ov::element::undefined) {
+                    if (dtype == Common::type_helpers::get_ov_type(array)) {
+                        return Common::object_from_data<ov::Tensor>(array, share);
+                    }
+                    // If dtypes does not match, create a copy:
+                    return Common::object_from_data<ov::Tensor>(array, dtype, false);
+                }
+                // Otherwise deduce dtype from an array:
+                return Common::object_from_data<ov::Tensor>(array, share);
+            }),
+            py::arg("array"),
+            py::kw_only(),
+            py::arg("dtype") = ov::element::undefined,
+            py::arg("share") = false,
+            R"(
+                Tensor's constructor.
+
+                :param array: Array to create the tensor from.
+                :type array: numpy.array
+                :param dtype: Desired type of Tensor. Leave undefined to inherit the dtype
+                              from numpy array. Casting will always result in copy!
+                              i.e. when creating Type.bf16 tensors or creating Type.f32 tensor
+                              from numpy.float64 array, data sharing will be disabled by defult.
```

Comment:
I'm against of tensor to convert data.
Data conversion is part of graph execution process.

Python bindings are allowed only to perform bind C++ and Python APIs, but not perform extra computations outside of main `infer`

Reply:
From NNCF's perspective, it might be very useful to have methods for data conversion somewhere in Python API.

Reply:
@ilya-lavrenov so what is you proposal to address it in OpenVINO? How should `bf16` data be correctly interpreted (from any possible array i.e. int16/float16/float32, with regards to values and not it's binary representation) and populate Tensors?

Reply:
I don't see the full problem from the tickets. They just provide some code which doesn't work, but I don't understand how it's related to the fact that we store bf16 as fp16 and we need to change this approach.